ArchUnitNET/Domain/PlantUml/Export/PlantUmlFileBuilder.cs

```diff
@@ -172,11 +217,319 @@ public PlantUmlFileBuilder WithDependenciesFrom(IEnumerable<Slice> slices, Gener
+                    targetSlice.Description != slice.Description &&
+                    slice.Dependencies.Where(dep =>
+                            generationOptions.DependencyFilter?.Invoke(dep) ?? true)
+                        .Any(dep => targetSlice.Types.Contains(dep.Target)));
+
+                if (sliceIsPackage)
+                {
+                    _dependencies.AddRange(dependencyTargets.Select(target =>
+                        new PlantUmlDependency(slice.Description, target.Description, DependencyType.PackageToOne)));
+                }
+                else
+                {
+                    _dependencies.AddRange(dependencyTargets.Select(target =>
+                        new PlantUmlDependency(slice.Description, target.Description, DependencyType.OneToOne)));
+                }
+
+                if (slice is Namespace) //This throws errors if namespaces and slices are used in the same diagram, the syntax can't be mixed in PlantUML
+                {
+                    nodes.Add(slice, new PlantUmlNamespace(slice.Description));
+                }
+                else if (!sliceIsPackage)
+                {
+                    nodes.Add(slice,
+                        slice.Description.Contains(package)
+                            ? new PlantUmlSlice(slice.Description, slice.NameSpace, "99ffd1")
+                            : new PlantUmlSlice(slice.Description, slice.NameSpace));
+                }
+                else
+                {
+                    if (!slice.Description.Contains(package))
+                    {
+                        nodes.Add(slice, 
+                            new PlantUmlSlice(slice.Description + ".", slice.NameSpace));
+                    }
+                }
+            }
+
+            foreach (var entry in nodes.Where(node =>
+                         !_dependencies.SelectMany(dep => new[] {dep.Origin, dep.Target})
+                             .Contains(node.Key.Description)))
+            {
+                nodes.Remove(entry.Key);
+            }
+
+            RemoveDuplicateDependenciesWhenShowingPackages();
+            ReplaceCirclesWithAppropriateDependencyType();
+            RemoveDuplicatedArrowsIfExist();
+
+            for (var j = _dependencies.Count - 1; j >= 0; j--)
+            {
+                if (!_dependencies[j].Origin.Contains(package) && !_dependencies[j].Target.Contains(package))
+                {
+                    _dependencies.RemoveAt(j);
+                }
+            }
+
+            for (var i = nodes.Count - 1; i >= 0; i--)
+            {
+                var nodeExist = _dependencies.Any(dep => nodes.ElementAt(i).Key.Description == dep.Origin || nodes.ElementAt(i).Key.Description == dep.Target);
+                if (!nodeExist)
+                {
+                    nodes.Remove(nodes.ElementAt(i).Key);
+                }
+            }
+
+            var nodeElements = nodes.Values.ToList();
+
+            foreach (var node in nodes)
```

Comment:
This `foreach`, for example, is identical to the one in `GetDependenciesFrom` and should be extracted.